File: Service/Translators/BadWeatherDayTranslator.vb
Imports BO
Imports DAL

Public Class BadWeatherDayTranslator
    Public Shared Function TranslateEntityToBO(_entity As BadWeatherDays, bo As BadWeatherDayBO) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        bo.Id = _entity.Id
        bo.WeatherStationId = _entity.WeatherstationId
        bo.Type = _entity.Type
        bo.BWDate = _entity.Date
        Return ErrorCode.Success
    End Function

    Friend Shared Function TranslateBOToEntity(_entity As BadWeatherDays, bo As BadWeatherDayBO) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        _entity.WeatherstationId = bo.WeatherStationId
        _entity.Type = bo.Type
        _entity.Date = bo.BWDate

        Return ErrorCode.Success
    End Function
End Class

File: ServiceCore/Translators/BadWeatherDayTranslator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BOCore;
using DALCore.Models;

namespace ServiceCore.Translators
{
    public class BadWeatherDayTranslator
    {
        public static ErrorCode TranslateEntityToBO(BadWeatherDays _entity, BadWeatherDayBO bo)
        {
            if (_entity == null)
                return ErrorCode.EntityNull;
            if (bo == null)
                return ErrorCode.BoNull;
            bo.Id = _entity.Id;
            bo.WeatherStationId = _entity.WeatherstationId;
            bo.Type = _entity.Type;
            bo.BWDate = _entity.Date;
            return ErrorCode.Success;
        }

        internal static ErrorCode TranslateBOToEntity(BadWeatherDays _entity, BadWeatherDayBO bo)
        {
            if (_entity == null)
                return ErrorCode.EntityNull;
            if (bo == null)
                return ErrorCode.BoNull;
            _entity.WeatherstationId = bo.WeatherStationId;
            _entity.Type = bo.Type;
            _entity.Date = bo.BWDate;

            return ErrorCode.Success;
        }
    }
}
